cdap-api/src/main/java/io/cdap/cdap/api/service/http/HttpServiceContext.java

```diff
@@ -79,6 +81,39 @@
    *
    * @param namespace the namespace for user scoped plugins
    * @return an dynamic plugin configurer that must be closed
+   * @deprecated use {@link #createServicePluginConfigurer(String)} instead
    */
+  @Deprecated
   PluginConfigurer createPluginConfigurer(String namespace);
+
+  /**
+   * Create a {@link HttpServicePluginConfigurer} that can be used to instantiate plugins at runtime that
+   * were not registered at configure time. Plugins registered by the dynamic configurer live in their own scope
+   * and will not conflict with any plugins that were registered by the service when it was configured.
+   * Plugins registered by the returned configurer will also not be available through
+   * {@link #newPluginInstance(String)} or {@link #newPluginInstance(String, MacroEvaluator)}.
+   * Plugins with system scope and plugins in the same namespace as the running service will be visible.
+   *
+   * The dynamic configurer is meant to be used to create plugins for the lifetime of a single service call and
+   * then to be forgotten.
+   *
+   * @return an dynamic plugin configurer that must be closed
+   */
+  HttpServicePluginConfigurer createServicePluginConfigurer();
```

Comment:
nit: rename HttpServicePluginConfigurer -> ServicePluginConfigurer

Reply:
Do we support this in cdap-api? I thought we only support dynamic plugin in service for system app services (cdap-sysmte-app-api)?

Reply:
We support it for general services, any service is able to load the plugins, I believe from 6.1.